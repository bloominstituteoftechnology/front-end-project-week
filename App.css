.App-left {
    background-color: #F3F3F3;
    float: left;
    width: 50%;
}

.App-right {
    background-color: grey;
    margin-left: 50%;
}


.App {
  text-align: left;
}



.App-header {
  background-color: dark grey;
  height: 150px;
  width: 20%;
  padding: 20px;
  color: black;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.box {
height:200px;
width: 200px;
border: 1px solid black;
color: black;
background-color: F3F3F3


}