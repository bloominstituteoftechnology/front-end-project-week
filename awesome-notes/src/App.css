html {
  font-size: 62.5%;
}

body {
  margin: 0;
  padding: 0;
  font-family: "Raleway", sans-serif;
  font-size: 1.6rem;
  color: #4a4a4a;
}

button {
  color: #ffffff;
}

h1,
h2,
h3,
h4,
h5 {
  font-family: "Roboto", sans-serif;
}

.App {
  position: relative;
  left: 110px;
}
