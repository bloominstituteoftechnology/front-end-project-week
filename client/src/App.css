.App {
  height: 100%; 
  margin: 0; 
  min-height: 100%; 
  width: 100%; 
}