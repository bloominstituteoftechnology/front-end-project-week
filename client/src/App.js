import React, { Component } from 'react';
import SideNav from './components/SideNav';
import styled from 'styled-components';
import { Route, Switch, withRouter, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import ListContainer from './containers/ListContainer';
import NoteContainer from './containers/NoteContainer';
import AddNoteContainer from './containers/AddNoteContainer';
import EditNoteContainer from './containers/EditNoteContainer';
import SearchBar from './containers/SearchBar';
import SortingOptions from './containers/SortingOptions';
import LoggedIn from './containers/LoggedIn';
import Register from './containers/Register';
import Login from './containers/Login';
import Fade from './components/Fade';
import { checkToken, logout } from './actions';

const MainContent = styled.div`
  padding: 4rem;
`;

class App extends Component {
  componentDidMount() {
    this.props.checkToken();
  }

  render() {
    return (
      <div>
        <LoggedIn>
          {props => <SideNav {...props} logout={this.props.logout} />}
        </LoggedIn>
        <MainContent>
          <Fade>
            {({ location }) => (
              <Switch location={location}>
                <Route
                  exact
                  path="/"
                  render={() =>
                    !this.props.isLoggedIn ? (
                      <Login />
                    ) : (
                      <Redirect to="/notes" />
                    )
                  }
                />
                <Route
                  exact
                  path="/register"
                  render={() =>
                    !this.props.isLoggedIn ? (
                      <Register />
                    ) : (
                      <Redirect to="/notes" />
                    )
                  }
                />
                <Route
                  exact
                  path="/notes"
                  render={() =>
                    this.props.isLoggedIn ? (
                      <React.Fragment>
                        <SearchBar />
                        <SortingOptions />
                        <ListContainer />
                      </React.Fragment>
                    ) : (
                      <Redirect to="/" />
                    )
                  }
                />
                <Route
                  exact
                  path="/notes/new"
                  render={props =>
                    this.props.isLoggedIn ? (
                      <AddNoteContainer {...props} />
                    ) : (
                      <Redirect to="/" />
                    )
                  }
                />
                <Route
                  exact
                  path="/notes/:id/edit"
                  render={props =>
                    this.props.isLoggedIn ? (
                      <EditNoteContainer {...props} />
                    ) : (
                      <Redirect to="/" />
                    )
                  }
                />
                <Route
                  exact
                  path="/notes/:id"
                  render={props =>
                    this.props.isLoggedIn ? (
                      <NoteContainer {...props} />
                    ) : (
                      <Redirect to="/" />
                    )
                  }
                />
              </Switch>
            )}
          </Fade>
        </MainContent>
      </div>
    );
  }
}

export default withRouter(
  connect(
    ({ isLoggedIn }) => ({ isLoggedIn }),
    { checkToken, logout },
  )(App),
);
