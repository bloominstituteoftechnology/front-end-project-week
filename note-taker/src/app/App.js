import React from "react";
import styled from "styled-components";
import { Route } from "react-router-dom";
import SidebarContainer from "./common/Sidebar/SidebarContainer";
import HomeContainer from "./home/HomeContainer";
import "./styles/custom-props.css";
import CreateNoteContainer from "./create-new/CreateNoteContainer";
import NoteContainer from "./note-view/NoteContainer";
import EditNoteContainer from "./edit-view/EditNoteContainer";
const AppContainer = styled.div`
  display: flex;
  color: var(--main-font-color);
  background: var(--main-bg);
  height: 100vw;
  `;
const MainContentContainer = styled.div`
  position: absolute;
  background: var(--main-bg);
  left: var(--sidebar-width);
  right: 0;
  padding: 60px 25px;
`;
const App = () => {
  return (
    <AppContainer>
      <SidebarContainer className = 'sidebar'/>
      <MainContentContainer className = 'main-content-container'>
        <Route exact path="/" component={HomeContainer} />
        <Route exact path="/create-new" component={CreateNoteContainer} />
        <Route exact path="/note/get/:id" component={NoteContainer} />
        <Route exact path="/:id/edit-view" component={EditNoteContainer} />
      </MainContentContainer>
    </AppContainer>
  );
};

export default App;
