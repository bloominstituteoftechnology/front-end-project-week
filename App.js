import React, { Component } from 'react';
/*import logo from './logo.svg'; */
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
         </header>
          <section className="App-section">
          <h1 className="App-title">Lambda Notes</h1> 
       </section>
       
      </div>
    );
  }
}

export default App;
