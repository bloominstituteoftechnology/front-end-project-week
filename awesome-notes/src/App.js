import React, { Component } from "react";
import "./App.css";

////////////////
// COMPONENTS //
////////////////
import LeftSideBar from "./components/LeftSideBar/LeftSideBar";
import ContentWindow from "./components/ContentWindow/ContentWindow";

class App extends Component {
  render() {
    return (
      <div className="App container">
        <LeftSideBar />
        <ContentWindow />
      </div>
    );
  }
}

export default App;
