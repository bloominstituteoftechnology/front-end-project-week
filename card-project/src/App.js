import React from 'react';
import './App.css';
import { Route } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';

import PresentationMain from './components/PresentationMain';
import PresentationView from './components/PresentationView';
import EditNote from './components/EditNote';
import CreateNew from './components/CreateNew';
import LambdaSide from './components/LambdaSide';


function LeftContent() {
	return (
		<div className="LeftContent">
			<LambdaSide />
		</div>
	);
}

function RightContent(props) {

	return (
		<div className="RightContent">
			<PresentationMain />
		</div>
	);
}

function SplitPane(props) {
	return (
		<div className="SplitPane">
			<div className="SplitPane-left">{props.left}</div>
			<div className="SplitPane-right">{props.right}</div>
		</div>
	);
}
const Home = (props) => {
	return <SplitPane left={<LeftContent />} right={<RightContent notes={props.notes} />} />;
};
const New = (props) => {
	return <SplitPane left={<LambdaNewL />} right={<LambdaNewR />} />;
};
const View = (props) => {
	let id = props.match.params.id;
	return (
		<div style={{ height: '100%' }}>
			<SplitPane left={<LambdaNewL />} right={<ViewRight id={id} />} />
		</div>
	);
};
export const Edit = (props) => {
	let idE = props.match.params.idE;
	return (
		<div style={{ height: '100%' }}>
			<SplitPane left={<LambdaNewL />} right={<EditRight idE={idE} />} />
		</div>
	);
};
const Delete = (props) => {
	let id = props.match.params.id;

	return (
		<div style={{ height: '100%' }}>
			<SplitPane left={<LambdaNewL />} right={<ViewRight id={id} />} />
		</div>
	);
};

const ViewRight = (props) => {
	return (
		<div className="RightContent">
			<PresentationView id={props.id} />
		</div>
	);
};
const EditRight = (props) => {
	return (
		<div className="RightContent">
			<EditNote idE={props.idE} />
		</div>
	);
};

const LambdaNewL = (props) => {
	return (
		<div className="LeftContent">
			<LambdaSide />
		</div>
	);
};
const LambdaNewR = (props) => {
	return (
		<div className="RightContent">
			<CreateNew />
		</div>
	);
};


const App = (props) => {
	return (
		<div style={{ height: '100%' }}>
			<Route exact path="/" component={Home} />
			<Route path="/notes/new" component={New} />
			<Route path="/notes/view/:id" component={View} />
			<Route path="/notes/edit/:idE" component={Edit} />
			<Route path="/notes/view/delete/:idE" component={Delete} />
		</div>
	);
};

export default App;
