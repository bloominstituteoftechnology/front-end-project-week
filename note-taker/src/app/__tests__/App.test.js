import React from "react";
import { shallow } from "enzyme";
import App from "../App";
import toJSON from "enzyme-to-json";
describe("<App />", () => {
  it("renders without crashing", () => {
    shallow(<App />);
  });
  it("contains a <SidebarContainer /> and <MainContentContainer />", () => {
    const app = shallow(<App />);
    const sidebar = app.find(".sidebar");
    const mainContentContainer = app.find(".main-content-container");
    expect(sidebar.length).toBe(1);
    expect(mainContentContainer.length).toBe(1);
  });
  it("<MainContentContainer></MainContentContainer> should have 4 <Route />", () => {
    const app = shallow(<App />);
    const mainContentContainer = app.find(".main-content-container");
    expect(mainContentContainer.find("Route").length).toBe(4);
  });
  it("checks every <Route /> to see if there is a path prop in <MainContentContainer></MainContentContainer>", () => {
    const app = shallow(<App />);
    const mainContentContainer = app.find(".main-content-container");
    mainContentContainer.find("Route").forEach(route => {
      expect(route.prop("path")).toBeDefined();
    });
  });

  it("matches the snapshot", () => {
    const tree = shallow(<App />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
