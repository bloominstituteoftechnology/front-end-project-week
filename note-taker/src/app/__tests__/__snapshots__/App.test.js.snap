// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<styled.div>
  <Connect(SidebarComponent)
    className="sidebar"
  />
  <styled.div
    className="main-content-container"
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/create-new"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/note/get/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/:id/edit-view"
    />
  </styled.div>
</styled.div>
`;
